fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/item/v1/FabricItem.java

```diff
@@ -90,4 +90,28 @@ default Multimap<EntityAttribute, EntityAttributeModifier> getAttributeModifiers
 	default boolean isSuitableFor(ItemStack stack, BlockState state) {
 		return ((Item) this).isSuitableFor(state);
 	}
+
+	/**
+	 * Returns a leftover item stack after {@code stack} is consumed in a recipe.
+	 * (This is also known as "recipe remainder".)
+	 * For example, using a lava bucket in a furnace as fuel will leave an empty bucket.
+	 *
+	 * <p>This is a stack-aware version of {@link Item#getRecipeRemainder()}.
+	 *
+	 * <p>Overriding this method must follow the rule,
+	 * to avoid compatibility issues with other mods using recipe remainders,
```

Comment:
I requested a size of 1 myself to ensure that
1) mods that rely on it don't need to setCount themselves;
2) mods that try to use this system to leave > 1 remainder after a recipe can be yelled at because it's not something I think we should allow.

Reply:
In fact, we might as well add a check in FabricItemStack#getRecipeRemainder to enforce it.

Reply:
> mods that rely on it don't need to setCount themselves;

Why would they need to `setCount` if the item specifically wants to produce a remainder of eg. 3 items?

Reply:
It was more in the sense of passing a stack with size > 1 for some reason and then having to manually set the count to 1 afterwards in case the implementation just copied the stack without setting the count to 1.